services/dialog/v1.js

```diff
@@ -0,0 +1,342 @@
+  // Otherwise, once set username would never get reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+	
+  var services = cfenv.getAppEnv().services;
+
+  var username, password, sUsername, sPassword;
+  var service = cfenv.getAppEnv().getServiceCreds(/dialog/i);
+  
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+  
+  RED.httpAdmin.get("/service-dialog/vcap", function (req, res) {
+		res.json(service ? {bound_service: true} : null);
+  });  
+  
+  
+  // This is the Dialog Node. 
+  // The node supports six modes
+  // create a dialog from a dialog template
+  // delete a dialog : the Watson Dialog service Instance is limited to 10 dialog instances
+  // list - retrieves a list of the Dialogs
+  // start conversation - initiates a conversation with an existing conversation
+  // continue a conversation - continues with a started conversation
+  // get profile variable - GETs the profile variables associated with the dialog
+
+  function WatsonDialogNode (config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+   
+    this.log("Watson Developer Cloud Contribution - Dialog Node Instantiated") ;
+	
+    this.on("input", function (msg) {
+      this.log("Watson Developer Cloud Contribution - Input received");
+		
+      var message = "";
+      if (!msg.payload) {
+			 message = "Missing property: msg.payload";
+			 node.error(message, msg);
+			 return;
+      }
+
+      var username = sUsername || this.credentials.username;
+      var password = sPassword || this.credentials.password;
+      this.status({});  
+			
+      if (!username || !password) {
+        this.status({fill:"red", shape:"ring", text:"missing credentials"});  
+        message = 'Missing Watson Dialog service credentials';
+        this.error(message, msg);
+        return;
+      } 
+
+      var dialog = watson.dialog({
+        username: username,
+        password: password,
+        version: "v1"
+      });	  
+
+      var params = {};
+      if (config.mode === "create") {
+        performCreate(node,dialog,msg);
+      }
+      else if (config.mode === "delete") {
+        performDelete(node,dialog,msg,config);
+      }
+      else if (config.mode === "deleteall") {
+        performDeleteAll(node,dialog,msg,config);
+      }
+       else if (config.mode === "list") {
+        performList(node,dialog,msg);
+      } else if (config.mode === "startconverse" || config.mode === "converse" || config.mode === "getprofile") {
+          var dialogid = config.dialog;
+          var clientid = config.clientid;
+          var converseid = config.converse;
+
+          if (!dialogid || "" == dialogid) {
+            if (msg.dialog_params && "dialog_id" in msg.dialog_params) {
+              dialogid = msg.dialog_params["dialog_id"];
+            }	
+          }
+		  
+          if (!dialogid || "" == dialogid) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'message' is already declared in line 80.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2385269578/issues/source?bid=3225115&fileBranchId=3225116#l115)